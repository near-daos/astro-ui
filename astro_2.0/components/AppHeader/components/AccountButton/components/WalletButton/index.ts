export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
