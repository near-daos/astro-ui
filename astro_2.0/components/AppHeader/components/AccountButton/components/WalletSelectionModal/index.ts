export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
