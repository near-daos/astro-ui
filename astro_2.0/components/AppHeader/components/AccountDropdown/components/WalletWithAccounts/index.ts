export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
