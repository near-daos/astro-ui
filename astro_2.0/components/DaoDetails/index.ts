export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
