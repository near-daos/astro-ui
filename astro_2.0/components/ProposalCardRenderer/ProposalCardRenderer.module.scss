@import 'styles/mixins';
@import 'styles/typography';

.root {
  border-radius: 0 8px 8px 0;
  display: grid;
  grid-template-areas:
    'header'
    'letterHead'
    'proposal'
    'infoPanel';
  grid-template-columns: 1fr;

  @include tablet-landscape {
    grid-template-areas:
      'header header'
      'letterHead proposal'
      'infoPanel infoPanel';
    grid-template-columns: 50px 1fr;
  }

  &.nonActionable * {
    pointer-events: none;
    user-select: none;
  }

  &.nonActionable .header {
    opacity: 0.2;
  }

  &.nonActionable .proposal::after {
    background-color: var(--color-white);
    border-radius: 8px;
    bottom: 0;
    content: '';
    cursor: default;
    left: 0;
    opacity: 0.8;
    position: absolute;
    right: 0;
    top: -48px;

    z-index: 10;

    @include tablet-landscape {
      left: -50px;
      top: 0;
    }
  }
}

.header {
  align-items: flex-end;
  display: flex;
  grid-area: header;
  justify-content: space-between;
}

.flagWrapper {
  margin-right: auto;
}

.proposal {
  grid-area: proposal;
  position: relative;
}

.letterHead {
  grid-area: letterHead;
}

.infoPanel {
  grid-area: infoPanel;
  padding-left: 8px;
  padding-right: 8px;

  @include tablet-landscape {
    padding-left: 19px;
    padding-right: 19px;
  }
}

.proposalIdCell {
  align-items: center;
  display: flex;
  height: 36px;
  white-space: nowrap;

  .proposalIdLabel {
    @extend %subtitle5;
    color: var(--color-neutral-60);
    font-weight: 400;
  }

  .proposalIdValue {
    @extend %subtitle5;
    align-items: center;
    color: var(--color-neutral-60);
    display: flex;
    gap: 12px;
  }
}

.saveDraftButton {
  margin-top: 16px;
}
