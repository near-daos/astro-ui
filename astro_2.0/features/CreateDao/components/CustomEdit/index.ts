import CustomEdit from './CustomEdit';

export default CustomEdit;
