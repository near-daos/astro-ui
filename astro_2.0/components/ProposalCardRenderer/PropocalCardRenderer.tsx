import React from 'react';
import cn from 'classnames';
import isNumber from 'lodash/isNumber';
import { useTranslation } from 'next-i18next';

import { InfoPanel } from 'astro_2.0/components/ProposalCardRenderer/components/InfoPanel';

import styles from './ProposalCardRenderer.module.scss';

export interface ProposalCardRendererProps {
  proposalId?: number | null;
  letterHeadNode?: React.ReactNode;
  proposalCardNode: React.ReactNode;
  daoFlagNode?: React.ReactNode;
  infoPanelNode?: React.ReactNode;
  className?: string;
  showInfo?: boolean;
  optionalActionNode?: React.ReactNode;
  isDraft?: boolean;
}

export const ProposalCardRenderer: React.FC<ProposalCardRendererProps> = ({
  letterHeadNode,
  daoFlagNode,
  proposalCardNode,
  infoPanelNode,
  className,
  proposalId,
  showInfo,
  optionalActionNode,
  isDraft,
}) => {
  const { t } = useTranslation();

  function renderFlag() {
    return daoFlagNode ? (
      <div className={styles.daoFlag}>{daoFlagNode}</div>
    ) : null;
  }

  function renderProposalId() {
    if (isDraft) {
      return null;
    }

    if (isNumber(proposalId)) {
      return (
        <div className={styles.proposalIdCell}>
          <span className={styles.proposalIdLabel}>
            {t('proposalCard.proposalID')}
          </span>
          <span className={styles.proposalIdValue}>{proposalId}</span>
        </div>
      );
    }

    return null;
  }

  function renderInfoPanel() {
    if (proposalId === undefined && showInfo) {
      return <InfoPanel />;
    }

    return null;
  }

  return (
    <div className={cn(styles.root, className)}>
      <div className={styles.header}>
        <div className={styles.flagWrapper}>{renderFlag()}</div>
        {renderInfoPanel()}
        {renderProposalId()}
        {optionalActionNode}
      </div>
      {letterHeadNode && (
        <div className={styles.letterHead}>{letterHeadNode}</div>
      )}
      <div className={styles.proposal}>{proposalCardNode}</div>
      {infoPanelNode && <div className={styles.infoPanel}>{infoPanelNode}</div>}
    </div>
  );
};
