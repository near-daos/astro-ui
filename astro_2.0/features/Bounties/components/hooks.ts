import { useCallback, useRef, useState } from 'react';
import { useRouter } from 'next/router';
import { useAsyncFn, useLocalStorage, useLocation } from 'react-use';
import axios, { CancelTokenSource } from 'axios';

import { SputnikHttpService } from 'services/sputnik';
import { useModal } from 'components/modal';
import { ConfirmActionModal } from 'astro_2.0/components/ConfirmActionModal';

import { Bounty, BountyContext, BountyProposal } from 'types/bounties';
import { VoteAction } from 'types/proposal';

import { NOTIFICATION_TYPES, showNotification } from 'features/notifications';

import { useWalletContext } from 'context/WalletContext';
import { PaginationResponse } from 'types/api';
import { VOTE_ACTION_SOURCE_PAGE } from 'constants/votingConstants';

export function useBountyControls(
  daoId: string,
  bounty?: Bounty
): {
  handleClaim: () => void;
  handleUnclaim: () => void;
  loading: boolean;
} {
  const router = useRouter();
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();
  const [loading, setLoading] = useState(false);

  const [showModal] = useModal(ConfirmActionModal);

  const onSuccessHandler = useCallback(async () => {
    router.reload();
  }, [router]);

  const handleClaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Claim',
    });

    if (res?.length && bounty) {
      setLoading(true);

      const dao = await SputnikHttpService.getDaoById(daoId);

      if (!dao) {
        return;
      }

      try {
        setVoteActionSource(pathname);

        await nearService?.claimBounty(daoId, {
          bountyId: bounty?.bountyId,
          deadline: bounty?.maxDeadline,
          bountyBond: dao.policy.bountyBond,
          gas: res[0],
          tokenId: bounty?.token,
        });

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    daoId,
    setVoteActionSource,
    pathname,
    nearService,
    onSuccessHandler,
  ]);

  const handleUnclaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Unclaim',
    });

    if (res?.length && bounty) {
      try {
        setLoading(true);
        setVoteActionSource(pathname);
        await nearService?.unclaimBounty(daoId, bounty?.bountyId);

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    setVoteActionSource,
    pathname,
    nearService,
    daoId,
    onSuccessHandler,
  ]);

  return {
    handleClaim,
    handleUnclaim,
    loading,
  };
}

export function useBountyVoting(
  daoId: string,
  proposal: BountyProposal
): {
  handleVote: (vote: VoteAction) => void;
  loading: boolean;
} {
  const router = useRouter();
  const [showModal] = useModal(ConfirmActionModal);
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();

  const [{ loading }, handleVote] = useAsyncFn(
    async (vote: VoteAction) => {
      const res = await showModal({
        title: 'Confirm Your Vote',
      });

      if (res?.length) {
        setVoteActionSource(pathname);
        await nearService?.vote(daoId, proposal.proposalId, vote, res[0]);
        await router.reload();
      }
    },
    [daoId, proposal, router, nearService, pathname, setVoteActionSource]
  );

  return {
    handleVote,
    loading,
  };
}

export function useBountySearch(): {
  handleSearch: (
    val: string
  ) => Promise<PaginationResponse<BountyContext[]> | null>;
  loading: boolean;
} {
  const router = useRouter();
  const daoId = router.query.dao as string;
  const { accountId } = useWalletContext();
  const cancelTokenRef = useRef<CancelTokenSource | null>(null);

  const [{ loading }, handleSearch] = useAsyncFn(async query => {
    if (cancelTokenRef.current) {
      cancelTokenRef.current?.cancel('Cancelled by new req');
    }

    const { CancelToken } = axios;
    const source = CancelToken.source();

    cancelTokenRef.current = source;

    return SputnikHttpService.findBountyContext({
      daoId: daoId || '',
      accountId,
      query,
      cancelToken: source.token,
    });
  }, []);

  return {
    handleSearch,
    loading,
  };
}

export function useHideBounty(): {
  handleSubmit: () => void;
  handleSelect: (id: string) => void;
  selected: string[];
  loading: boolean;
} {
  const router = useRouter();
  const showHidden = router.query?.bountyFilter === 'hidden';
  const daoId = router.query.dao as string;
  const { accountId, nearService, pkAndSignature } = useWalletContext();
  const [selected, setSelected] = useState<string[]>([]);

  const [{ loading }, handleSubmit] = useAsyncFn(async () => {
    if (!pkAndSignature) {
      return;
    }

    try {
      const { publicKey, signature } = pkAndSignature;

      if (publicKey && signature) {
        await SputnikHttpService.toggleBountyContexts({
          accountId,
          publicKey,
          signature,
          daoId,
          ids: selected,
          isArchived: !showHidden,
        });

        await router.replace(router.asPath, undefined, {
          shallow: false,
        });

        setSelected([]);
      }
    } catch (err) {
      showNotification({
        type: NOTIFICATION_TYPES.ERROR,
        lifetime: 20000,
        description: err.message,
      });
    }
  }, [
    selected,
    daoId,
    showHidden,
    accountId,
    router,
    nearService,
    pkAndSignature,
  ]);

  const handleSelect = useCallback(
    (id: string) => {
      if (selected.includes(id)) {
        setSelected(selected.filter(item => item !== id));
      } else {
        setSelected([...selected, id]);
      }
    },
    [selected]
  );

  return {
    handleSubmit,
    handleSelect,
    selected,
    loading,
  };
}
