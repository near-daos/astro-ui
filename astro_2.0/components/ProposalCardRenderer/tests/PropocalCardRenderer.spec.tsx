import { render } from 'jest/testUtils';

import { ProposalCardRenderer } from 'astro_2.0/components/ProposalCardRenderer';

describe('proposal card renderer', () => {
  it('Should render component', () => {
    const { container } = render(
      <ProposalCardRenderer proposalCardNode={<div>Proposal Node</div>} />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render dao flag node', () => {
    const daoFlagNode = 'Dao Flag Node';

    const { getAllByText } = render(
      <ProposalCardRenderer
        proposalCardNode={<div>Proposal Node</div>}
        daoFlagNode={daoFlagNode}
      />
    );

    expect(getAllByText(daoFlagNode)).toHaveLength(1);
  });

  it('Should render letter head node', () => {
    const letterHeadNode = 'Letter Head Node';

    const { getAllByText } = render(
      <ProposalCardRenderer
        proposalCardNode={<div>Proposal Node</div>}
        letterHeadNode={letterHeadNode}
      />
    );

    expect(getAllByText(letterHeadNode)).toHaveLength(1);
  });

  it('Should render info panel node', () => {
    const infoPanelNode = 'Info Panel Node';

    const { getAllByText } = render(
      <ProposalCardRenderer
        proposalCardNode={<div>Proposal Node</div>}
        infoPanelNode={infoPanelNode}
      />
    );

    expect(getAllByText(infoPanelNode)).toHaveLength(1);
  });
});
