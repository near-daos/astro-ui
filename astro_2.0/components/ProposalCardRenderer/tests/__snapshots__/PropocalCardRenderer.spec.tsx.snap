// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proposal card renderer Should render component 1`] = `
<div>
  <div
    class=""
  >
    <div>
      <div />
      <div
        class="undefined"
        data-testid="hover-el"
      >
        <div>
          <svg>
            <use
              href="/_next/undefined"
            />
          </svg>
        </div>
        <div>
          popupsInfoWarning
        </div>
      </div>
    </div>
    <div>
      <div>
        Proposal Node
      </div>
    </div>
  </div>
</div>
`;
